odl/operator/tensor_ops.py

```diff
@@ -337,7 +337,14 @@ def derivative(self, vf):
         inner_vf = vf.copy()
 
         for gi in inner_vf:
-            gi /= vf_pwnorm_fac * gi ** (self.exponent - 2)
+            if self.exponent >= 2:
+                tmp = vf_pwnorm_fac * gi ** (self.exponent - 2)
+                # Any component that is zero is not divided with
+                for i, val in zip(np.arange(len(tmp)), tmp):
```

Comment:
My previous attempt on this looked like
```python
tmp = vf_pwnorm_fac * gi ** (self.exponent - 2)
# Any component that is zero is not divided with
gi /= tmp if not tmp else 1.0
```
in which I hoped that the ``if``-``else`` statement would make a point-wise comparison and thus create a mask. However, when travis faild on this one i checked and that was not the case. @adler-j or @kohr-h : is there a better way to do this, instead of looping over the components?

Reply:
No, `if numpy_array` will fail because an array cannot be unambiguously treated as a boolean.

Use a boolean mask:
```python
nz = (tmp != 0)
gi[nz] /= tmp[nz]
```

Reply:
And cast to Numpy arrays before doing that, of course. (We really need to get the tensor branch in, then all this casting back and forth has an end...)